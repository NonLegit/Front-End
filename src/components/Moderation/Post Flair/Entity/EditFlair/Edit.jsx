import { useState } from 'react';
import { useParams } from 'react-router-dom';
import Done from '../../../../AlertMessage';
import EditFlair from './EditFlair';
import {
  AddFlair,
  Cancel,
  Color,
  ColorData,
  Container, Count, FieldData, FormCont, Input, Lable, Save, SecondLable,
} from './style';

/**
 * Edit Entity of Flair
 * @component
 * @property  {function} check copy id
 * @property  {function} notEmpty check if input feild is empty or not
 * @property  {function} SendData send data to backend
 * @property  {function} EditFlair send edited data to backend
 * @property  {function} save close popup form
 * @property  {function} cancel popup form
 * @property  {function} flair flaie entity

 * @return {React.Component} - Edit Entity of Flair
 */

export default function Edit(props) {
  const { flair, save, cancel } = props;
  const [count, setCount] = useState(64 - flair.text.length);
  const [error, setError] = useState('');
  const { subReddit } = useParams();

  const check = (e) => {
    const ele = e?.target?.value;
    setCount(64 - ele.length);
  };
  const notEmpty = (e) => {
    const ele = e?.target?.value;
    if (ele.length === 0) {
      setError('Error: text or emoji is required');
    } else {
      setError('');
    }
  };
  const SendData = () => {
    const text = document.getElementById('text').value;
    const backgroundColor = document.getElementById('bgColor').value;
    // const cssClass = document.getElementById('css');
    const textColor = document.getElementById('textColor').value;
    console.log(text, backgroundColor, textColor);

    EditFlair(subReddit, flair?._id, {
      text,
      backgroundColor,
      textColor,
    });
    save();
    Done('Chndes Save');
  };
  return (
    <>
      <Container>
        <Lable>FLAIR APPEARANCE</Lable>
        <FieldData>
          <SecondLable>Flair text</SecondLable>
          <FormCont>
            <Input
              id="text"
              onChange={check}
              defaultValue={flair?.text}
              onInput={(e) => {
              // eslint-disable-next-line radix
                e.target.value = e.target.value.slice(0, 21);
              }}
              onBlur={notEmpty}
            />
          </FormCont>
          {(error.length === 0)
            ? (
              <Count
                condition={(count === 0).toString()}
              >
                {count}
                {' '}
                Characters remaining
              </Count>
            )
            : <Count condition="true">{error}</Count>}
        </FieldData>

        <FieldData>
          <SecondLable>CSS Class</SecondLable>
          <FormCont>
            <Input
              id="css"
              onChange={check}
              defaultValue={flair?.cssClass}
              placeholder="none"
            />
          </FormCont>
          <Count>
            Optional
          </Count>
        </FieldData>

        <ColorData>
          <SecondLable>Flair background color</SecondLable>
          <Color
            type="color"
            id="bgColor"
            defaultValue={flair?.backgroundColor}
          />
        </ColorData>

        <ColorData>
          <SecondLable>Flair text color</SecondLable>
          <Color
            type="color"
            id="textColor"
            defaultValue={flair?.textColor}
            placeholder="Aa"
          />
        </ColorData>
      </Container>
      <AddFlair>
        <Save onClick={SendData}>
          Save
        </Save>
        <Cancel onClick={cancel}>Cancel</Cancel>
      </AddFlair>
    </>
  );
}
