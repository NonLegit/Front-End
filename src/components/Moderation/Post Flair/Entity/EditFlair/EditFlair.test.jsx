import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Edit from './Edit';

// test snapshot
const save = () => {};
const cancel = () => {};
test('test snapshot', async () => {
  const props = {
    flair: {
      text: 'aaa',
    },
    save,
    cancel,
  };
  const tree = renderer.create(<Edit {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
