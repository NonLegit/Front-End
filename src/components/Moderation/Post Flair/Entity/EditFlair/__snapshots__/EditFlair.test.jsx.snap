// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
Array [
  <div
    className="css-e2plnr"
  >
    <div
      className="css-17olsxy"
    >
      FLAIR APPEARANCE
    </div>
    <div
      className="css-7faeei"
    >
      <div
        className="css-1lg26qb"
      >
        Flair text
      </div>
      <div
        className="MuiFormControl-root css-11tsdoc-MuiFormControl-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1hrdv6q-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
          onInput={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            defaultValue="aaa"
            disabled={false}
            id="text"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        className="css-1jfe1hz"
      >
        61
         
        Characters remaining
      </p>
    </div>
    <div
      className="css-7faeei"
    >
      <div
        className="css-1lg26qb"
      >
        CSS Class
      </div>
      <div
        className="MuiFormControl-root css-11tsdoc-MuiFormControl-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1hrdv6q-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="css"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="none"
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        className="css-1jfe1hz"
      >
        Optional
      </p>
    </div>
    <div
      className="css-bg8uia"
    >
      <div
        className="css-1lg26qb"
      >
        Flair background color
      </div>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-dnud4j-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="bgColor"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="color"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="css-bg8uia"
    >
      <div
        className="css-1lg26qb"
      >
        Flair text color
      </div>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-dnud4j-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="textColor"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Aa"
          type="color"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-ihdtdm"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <div
    className="css-1gkf4ei"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1237fzv-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-4te1ww-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cancel
    </button>
  </div>,
]
`;
